### PR TITLE
OJ-2450: Added metrics

### PR DESCRIPTION
**Pull request by surajkumar**
### What changed and why
The `template.yaml` has been updated to include metrics. 

Metrics collected are:

- Metrics on successful NINO checks
- Metrics on retry sent
- Metrics on failed HMRC authentication
- Metrics on generic HMRC API errors
- Metrics on Matching Lambda exceptions
- Metrics on deceased users
- Metrics on attempts exceeded
- Metrics on invalid sessions
- Metrics on when a CI is raised
- Metrics on when a VC is issued

### Issue tracking
- [OJ-2450](https://govukverify.atlassian.net/browse/OJ-2450)

[OJ-2450]: https://govukverify.atlassian.net/browse/OJ-2450?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -1235,6 +1235,110 @@ Resources:
...
+          MetricNamespace: !Sub ${AWS::StackName}/InvalidSessionErrorMetric
+
+  CIRaisedMetric:
+    Type: AWS::Logs::MetricFilter
+    Properties:
+      LogGroupName: !Ref NinoIssueCredentialLogGroup
+      FilterPattern: '{($.details.name = "Fetch CI")}'
+      MetricTransformations:
+        - MetricValue: 1
+          MetricName: InvalidSessionErrorMetric
```
Comment (jkunle) [inline infrastructure/template.yaml:1330]: ??? is this right

File: infrastructure/template.yaml [hunk 2]
```diff
@@ -1235,6 +1235,110 @@ Resources:
...
+          MetricNamespace: !Sub ${AWS::StackName}/CIRaisedMetric
+
+  VCIssuedMetric:
+    Type: AWS::Logs::MetricFilter
+    Properties:
+      LogGroupName: !Ref NinoIssueCredentialLogGroup
+      FilterPattern: '{($.details.name = "Create Signed JWT")}'
+      MetricTransformations:
+        - MetricValue: 1
+          MetricName: InvalidSessionErrorMetric
```
Comment (jkunle) [inline infrastructure/template.yaml:1340]: ???

Comment (surajkumar) [reply infrastructure/template.yaml:1340]: Oops!

File: infrastructure/template.yaml (hunk 1 above)
Comment (surajkumar) [reply infrastructure/template.yaml]: Fixed

File: infrastructure/template.yaml (hunk 2 above)
Comment (surajkumar) [reply infrastructure/template.yaml]: Fixed

**Review by jkunle — APPROVED** [review]

**Pull request merged**